rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +83,293 @@ typedef struct
     RotationBatch rotation_batch;
 }JointsDataBatch;
 
+typedef struct MetaDataInfo {
+public:
+    uint _img_id = -1;
+    std::string _img_name = "";
+    ImgSize _img_size = {};
+} MetaDataInfo;
+
 struct MetaData
 {
-    int& get_label() { return _label_id; }
-    BoundingBoxCords& get_bb_cords() { return _bb_cords; }
-    BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { return _bb_cords_xcycwh; }
-    BoundingBoxLabels& get_bb_labels() { return _bb_label_ids; }
-    void set_bb_labels(BoundingBoxLabels bb_label_ids) {_bb_label_ids = std::move(bb_label_ids); }
-    ImgSize& get_img_size() { return _img_size; }
-    const JointsData& get_joints_data(){ return _joints_data; }
-protected:
-    BoundingBoxCords _bb_cords = {}; // For bb use
-    BoundingBoxCords_xcycwh _bb_cords_xcycwh = {}; // For bb use
-    BoundingBoxLabels _bb_label_ids = {};// For bb use
-    ImgSize _img_size = {};
-    JointsData _joints_data = {};
-    int _label_id = -1; // For label use only
+    virtual std::vector<int>& get_labels() { };
+    virtual void set_labels(Labels label_ids) { };
+    virtual BoundingBoxCords& get_bb_cords() { };
+    virtual BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { };
+    virtual void set_bb_cords_xcycwh(BoundingBoxCords_xcycwh bb_cords_xcycwh) { };
+    virtual void set_bb_cords(BoundingBoxCords bb_cords) { };
+    virtual std::vector<int>& get_polygon_count() { };
+    virtual std::vector<std::vector<int>>& get_vertices_count() { };
+    virtual MaskCords& get_mask_cords() { };
+    virtual void set_mask_cords(MaskCords mask_cords) { };
+    virtual void set_polygon_counts(std::vector<int> polygon_count) { };
+    virtual void set_vertices_counts(std::vector<std::vector<int>> vertices_count) { };
+    virtual JointsData& get_joints_data(){  };
+    ImgSize& get_img_size() {return _info._img_size; }
+    std::string& get_image_name() { return _info._img_name; }
+    uint& get_image_id() { return _info._img_id; }
+    void set_img_size(ImgSize img_size) { _info._img_size = std::move(img_size); }
+    void set_img_id(uint img_id) { _info._img_id = img_id; }
+    void set_img_name(std::string img_name) { _info._img_name = std::move(img_name); }
+    void set_metadata_info(MetaDataInfo info) { _info = std::move(info); }
+    protected:
+    MetaDataInfo _info;
 };
 
 struct Label : public MetaData
 {
-    Label(int label) { _label_id = label; }
-    Label(){ _label_id = -1; }
+    Label(int label)
+    {
+        _label_ids.resize(1);
+        _label_ids[0] = label;
+    }
+    Label()
+    {
+        _label_ids.resize(1);
+        _label_ids[0] = -1;
+    }
+    std::vector<int>& get_labels() { return _label_ids; }
+    void set_labels(Labels label_ids)
+    {
+        _label_ids = std::move(label_ids);
+    }
+    protected:
+    Labels _label_ids = {}; // For label use only
```

Comment:
Can we initialize this as

```suggestion
    Labels _label_ids = {}; // For label use only
```
`Labels _label_ids = {-1};`


So in lines 123 and 128 we need not do a resize and we can initialize always with a vector with size as 1
But just a single line change as `_label_ids = label;`

Reply:
Done